### ISSUE TITLE
`Stripeldentity.onVerificationFailed` Crash error for Android users

### ISSUE THREAD

**Issue by cjmisenas-stripe** [capacitor-community/stripe#368]
**Platform**

- [ ] Web
- [ ] iOS
- [x] Android

**Describe the bug**
Seeing a frequent crash happen with the following stack trace for `5.4.2` on Android
```
Stripeldentity.onVerificationFailed
java.lang.NullPointerException - Attempt to invoke virtual method 'void com.getcapacitor. PluginCall.resolve(com.getcapacitor. JSObject) on a null object reference
```

**To Reproduce**

**Expected behavior**

**Screenshots**

**Additional context**

**Usage Product**
If you adopt this plugin, please let us know which product you are using.

Product Name: 
Product URL :
Using Function:
- [ ] Payment Sheet / Payment Flow
- [ ] Apple Pay
- [ ] Google Pay
- [x] Identity (@capacitor-community/stripe-identity)
- [ ] Terminal（@capacitor-community/stripe-terminal）

**Comment by rdlabo** [comment]
I checked with Demo and onVerificationFailed did not cause a Crash error. There seems to be another cause. Please give me a small reproduction code or step where the problem occurs.

**Comment by rdlabo** [comment]
This Issue is closed due to a long period of inactivity. If you still have issues, please open a new Issue. Best regards.

**Comment by yankostadinov** [comment]
@rdlabo 
This issue has been causing crashes in our android app continuously for months now.

It seems to be happening because the `call` variable is `null` in this file:
https://github.com/capacitor-community/stripe/blob/7b3a81ea646417da0b4dbcb3a2c9cdde43a2700f/packages/identity/android/src/main/java/com/getcapacitor/community/stripe/identity/StripeIdentity.java#L75-L76

The `call` variable is supposed to be setup in the`bridge`, but there is no saved call.
https://github.com/capacitor-community/stripe/blob/7b3a81ea646417da0b4dbcb3a2c9cdde43a2700f/packages/identity/android/src/main/java/com/getcapacitor/community/stripe/identity/StripeIdentityPlugin.java#L52

The error seems to happening within the first 5 seconds (as per Crashlytics) of a user's session in our app. So it seems to be a crash happening during initialization of the plugin or when resuming an activity with the plugin, which could help with reproducing it (I haven't been able to do so myself yet).

For what it's worth, we have had reports for users (and session recordings) that indicate this issue could be more common on Motorola devices, although most of the crashlytics reports are on Samsung devices still.

error:
```
Fatal Exception: java.lang.RuntimeException
Unable to resume activity {***********/**********.MainActivity}: java.lang.RuntimeException: Failure delivering result ResultInfo{who=null, request=2012565167, result=0, data=null} to activity {*********/********.MainActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'void com.getcapacitor.PluginCall.resolve(com.getcapacitor.JSObject)' on a null object reference
```
call stack:
```
android.app.ActivityThread.performResumeActivity (ActivityThread.java:5106)
android.app.ActivityThread.handleResumeActivity (ActivityThread.java:5139)
android.app.servertransaction.ResumeActivityItem.execute (ResumeActivityItem.java:57)
android.app.servertransaction.ActivityTransactionItem.execute (ActivityTransactionItem.java:45)
android.app.servertransaction.TransactionExecutor.executeLifecycleState (TransactionExecutor.java:180)
android.app.servertransaction.TransactionExecutor.execute (TransactionExecutor.java:98)
android.app.ActivityThread$H.handleMessage (ActivityThread.java:2567)
android.os.Handler.dispatchMessage (Handler.java:111)
android.os.Looper.loopOnce (Looper.java:242)
android.os.Looper.loop (Looper.java:362)
android.app.ActivityThread.main (ActivityThread.java:8448)
java.lang.reflect.Method.invoke (Method.java)
com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:552)
com.android.internal.os.ZygoteInit.main (ZygoteInit.java:992)
```

**Comment by manpreet-stripe** [comment]
[rdlabo](https://github.com/rdlabo)checking if you were able to see the above call stack/trace that is causing crashes in android app

**Comment by rdlabo** [comment]
@yankostadinov Thanks for information. Now we can finally discuss it!

This alone makes me think that `IdentityVerificationSheet.Companion` is being executed more than once.
Can you check the number of times apps has been executed? (I think the quickest way is to check in the Log)

Also make sure that present() is not mistakenly executed twice.

**Comment by rkarroll** [comment]
@rdlabo I work with @yankostadinov and we'll get back to you soon!

**Comment by Julian-Sam** [comment]
@rdlabo what do you mean by `number of times apps has been executed`?

**Comment by Julian-Sam** [comment]
Hey @rdlabo, we still have not been able to reproduce this, and its still continuing to affect our users as per our crash logs. We see that users app crashes on app startup. I wanted to propose that we gracefully handle the scenario when the `call` variable is `null`. This would help stop these crashes, especially since it happens at app startup, theres no reason for the app to crash altogether here when the user might not even be going through the Stripe flow.. wdyt?

**Comment by manpreet-stripe** [comment]
@rdlabo would you need additional details to help fix the crash?

**Comment by kentwilliams-stripe** [comment]
Hi @Julian-Sam @rdlabo I've investigated this issue as best I can (full disclaimer, I am not familiar at all with Capacitor's native bridge system) and I've made a PR that will hopefully fix this https://github.com/capacitor-community/stripe/pull/420. Please let me know your feedback and whether or not this will fix this issue.

Also, @yankostadinov I noticed that you referenced an older commit with Java source, do we need to also update an older version of this package?

**Comment by rdlabo** [comment]
Try this to see if it fixes it. If you get the same error with this, it seems to be returning multiple results.

https://github.com/capacitor-community/stripe/pull/426

**Comment by rdlabo** [comment]
Apologies—I forgot to follow up here after #426 was merged.

#426 fixed the reported crash by removing the assumption that the original saved `PluginCall` would still exist when Stripe Identity delivered its result. Thank you for the detailed production information, especially the Activity-resume stack trace and the confirmation that `bridge.getSavedCall(callbackId)` was returning `null`; that was the key to understanding the lifecycle failure.

We have also opened #488 to complete the result-recovery path. If Identity delivers `VerificationResult` after the Activity and JavaScript runtime have been recreated, but before the new JavaScript runtime has registered its listener, the plugin will retain the result until that listener is registered during application bootstrap.

This is an in-memory handoff and does not restore the old JavaScript execution context or guarantee recovery after full OS process death, but it prevents the result from being lost in the Activity-recreation path described here.
